### PR TITLE
Actions: Link checker

### PR DESCRIPTION
**Pull request by codeallthethingz**
This PR adds a check to the monty.yml.
It runs the docs link checker (tools.github_readme_sync check docs) if there are any changes to `/docs/*` and fails if there are broken links and outputs the broken links in the action output.

image

### CODE REVIEW

File: docs/overview/welcome-to-the-thousand-brains-project-documentation.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ---
 title: Welcome to the Thousand Brains Project Documentation!
 ---
-
+
```
Comment (tristanls) [inline docs/overview/welcome-to-the-thousand-brains-project-documentation.md:4]: **issue**: Please don't add extra whitespace.

I found this VSCode extension to be helpful with trailing whitespace: https://marketplace.visualstudio.com/items?itemName=shardulm94.trailing-spaces

File: docs/overview/welcome-to-the-thousand-brains-project-documentation.md
Comment (tristanls) [inline docs/overview/welcome-to-the-thousand-brains-project-documentation.md:1]: **question**: I assume this docs change is here to trigger a full docs workflow execution?

File: .github/workflows/monty.yml [hunk 2]
```diff
@@ -277,6 +330,52 @@ jobs:
           path: |
             ~/miniconda
           key: ${{ steps.restore_cache.outputs.cache-primary-key }}
+          
+  should_run_docs:
```
Comment (tristanls) [inline .github/workflows/monty.yml:334]: **issue**: By our convention, this name should end with `_monty`, and the job name below should be `name: should-run-docs-monty`.

**Review by tristanls — COMMENTED** [review]
Thanks! It looks like a great start toward automating our docs deployment 👍.

**issue**: This workflow looks independent of the Monty workflow. I think it should be modeled as a separate Docs workflow.

This would give us:
- Tools workflow runs for changes in `tools/*` and its workflow definition.
- Monty workflow runs for code changes and its workflow definition.
- Docs workflow would run for changes in `docs/*`, its workflow definition, and I think it should also run when `tools/github_readme_sync/*` changes.

Our GitHub Action naming convention:

- workflow name: Human-readable descriptive Captialized Case, e.g., "Docs", "Monty", "Tools", "Potato Stuff"
- job name: Human-readable kebab case ending with `*-<workflow-name>`, e.g., "check-docs", "install-monty", "test-tools", "check-style-potato-stuff"

**Comment by codeallthethingz** [comment]
Closing to refactor over on my fork.

File: docs/overview/welcome-to-the-thousand-brains-project-documentation.md
Comment (codeallthethingz) [reply docs/overview/welcome-to-the-thousand-brains-project-documentation.md:1]: yes.

**Comment by codeallthethingz** [comment]
@tristanls ok, I added the status check to the require ruleset -

image

File: .github/workflows/docs.yml [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+        if: ${{ github.event_name == 'pull_request' }}
+        id: filter_by_path_pr
+        working-directory: tbp.monty
+        run: |
+          git diff --name-only origin/${{ github.base_ref }} > changed_files.txt
+          while IFS= read -r changed_file
+          do
+            echo $changed_file
+            if [[ $changed_file == docs/* ]] ||
+               [[ $changed_file == tools/github_readme_sync/* ]]; then
```
Comment (tristanls) [inline .github/workflows/docs.yml:91]: **issue**: I forgot to mention before that the check should also run when the workflow file changes. See https://github.com/thousandbrainsproject/tbp.monty/blob/0d33c9c43123464ee0d2ba109cd734bba823bb9d/.github/workflows/tools.yml#L38

**Review by tristanls — APPROVED** [review]
Looks great! I forgot to mention one other condition when the check will run. Please update it before merging.

**Comment by tristanls** [comment]
Another request, please add `if: ${{ github.repository_owner == 'thousandbrainsproject' }}` to `should_run`.

File: .github/workflows/docs.yml (hunk 3 above)
Comment (codeallthethingz) [reply .github/workflows/docs.yml]: Ah this would have allowed me to not have to change a /doc file to get this to run too.  Smart.

**Comment by tristanls** [comment]
> This one I removed on purpose because I want this check to run on my fork and it doesn't require any secrets.

I agree that to test, it is useful to omit. However, because this workflow triggers `on: push`, I think it will cause errors on people's forks when they update them. This was what caused me to add the `if` condition.

**Pull request merged**